File: javascript/problem055.js
function isPalindrome(n){
  var str = n.toString();
  for(a = 0; a < parseInt(str.length/2) + 1; a++){
    if(str.substring(a, a+1) != str.substring(str.length - a - 1, str.length - a)){
      return false
    }
  }
  return true;
}


function getReverse(n){
    var reverse = ""
    var str = n.toString();
    for(a=0; a<str.length; a++){
        reverse += str.substring(str.length - a - 1, str.length - a);
    }
    return parseInt(reverse);
}


function isLychrel(n){
    var value = n
    var reverseValue = getReverse(value)
    var count = 1
    while(isPalindrome(value + reverseValue) == false){
        value = value + reverseValue
        reverseValue = getReverse(value)
        count += 1
        if(count == 50){
           return true;
        }
    }
    return false;
}


var result = 0
for (i=1; i<10000; i++){
    if(isLychrel(i)){
      result = result + 1;
    }
}

console.log(result)

File: java/src/Problem055.java
import javafx.util.converter.BigIntegerStringConverter;

import java.math.BigInteger;

/**
 * Created by BK on 2016-04-03.
 */
public class Problem055 {

    private BigIntegerStringConverter bigIntegerStringConverter = new BigIntegerStringConverter();

    public static void main(String args[]) {
        Problem055 problem = new Problem055();
        System.out.println(problem.result());
    }

    int result() {
        int value = 0;
        for (int i = 1; i <= 10000; i++) {
            if (isLychrel(BigInteger.valueOf(i))) {
                value = value + 1;
            }
        }
        return value;
    }

    boolean isPalindrome(BigInteger n) {
        String str = n.toString();
        for (int a = 0; a < (str.length() / 2) + 1; a++) {
            if (Integer.parseInt(str.substring(a, a + 1)) != Integer.parseInt(str.substring(str.length() - a - 1, str.length() - a))) {
                return false;
            }
        }
        return true;
    }

    boolean isLychrel(BigInteger n) {
        BigInteger original = n;
        BigInteger reverse = getReverse(n);
        int count = 1;
        while(isPalindrome(original.add(reverse)) == false){
            original = original.add(reverse);
            reverse = getReverse(original);
            count = count + 1;
            if(count == 50){
                return true;
            }
        }
        return false;
    }

    BigInteger getReverse(BigInteger n) {
        String str = n.toString();
        String reverseString = "";
        for (int a = str.length(); a > 0; a--) {
            reverseString = reverseString + str.substring(a - 1, a);
        }
        return bigIntegerStringConverter.fromString(reverseString);
    }


}
